src/storage/page_buffer.c

```diff
@@ -2361,7 +2361,7 @@ pgbuf_check_page_ahead_of_replication (THREAD_ENTRY * thread_p, PAGE_PTR page)
   if (page_lsa > repl_lsa)
     {
       LOG_TDES *tdes = LOG_FIND_CURRENT_TDES (thread_p);
-      assert (tdes->page_desync_lsa.is_null ());
+      assert (tdes != nullptr && !tdes->page_desync_lsa.is_null ());
```

Comment:
It is here that `page_desync_lsa` is being set. It must be null.

```suggestion
      assert (tdes != nullptr && tdes->page_desync_lsa.is_null ());
```

Nit.
Above, add `const`:
```
LOG_TDES *const tdes = LOG_FIND_CURRENT_TDES (thread_p);
```
.. and elsewhere around this PR.